pkg/resourcemanager/keyvaults/keyvault.go

```diff
@@ -210,9 +211,14 @@ func ParseAccessPolicy(policy *v1alpha1.AccessPolicyEntry, ctx context.Context)
 	}
 
 	if policy.ClientID != "" {
-		if objID := getObjectID(ctx, policy.TenantID, policy.ClientID); objID != nil {
-			newEntry.ObjectID = objID
+		objID, err := getObjectID(ctx, policy.TenantID, policy.ClientID)
+		if err != nil {
+			return keyvault.AccessPolicyEntry{}, err
```

Comment:
It is generally not a safe bet to check a return value when err is not nil. We could change the return value to something else but I wonder what we would put in it and what value it would add?